client/layout/global-sidebar/main.jsx

```diff
@@ -31,8 +31,7 @@ const GlobalSidebar = ( {
 	const isDesktop = useBreakpoint( '>=782px' );
 	const previousRoute = useSelector( getPreviousRoute );
 	const section = useSelector( getSection );
-	const previousLink = useRef( previousRoute );
-
+	const [ previousLink, setPreviousLink ] = useState( '' );
```

Comment:
Question: So the previous implementation didn't work before it wasn't updated on re-render due to being a ref?

Reply:
There were 2 issues
* The default value was configured to the `previousRoute` so the value became the `/me/account` page after the language was changed.
* The link of the `a` element couldn't get the latest value because it was a ref. This led to the link always pointing to `/sites` even if you went to the domains page and then the account page.